### PR TITLE
Implemented if statement, unary op "not", binary op "geq" & "leg"

### PR DESCRIPTION
**Pull request by SasiTMM**

### CODE REVIEW

File: Output/PSIOutput.il [hunk 1]
```diff
@@ -0,0 +1,243 @@
+.assembly extern System.Runtime { .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A) .ver 7:0:0:0 }
+.assembly extern System.Console { .publickeytoken = (B0 3F 5F 7F 11 D5 0A 3A) .ver 7:0:0:0 }
+.assembly extern PSILib { .ver 1:0:0:0 }
+.assembly Comp4 { .ver 0:0:0:0 }
```
Comment (tarydon) [inline Output/PSIOutput.il:4]: PSIOutput.dll, PSIOutput.il should not be part of the commit - these files are generated by the compiler and will keep changing every time. Exclude them by adding them to your .gitignore

File: PSICompile/ILCodeGen.cs [hunk 2]
```diff
@@ -67,7 +67,17 @@ public class ILCodeGen : Visitor {
...
+      string lab1 = NextLabel (), lab2 = NextLabel (), lab3 = NextLabel ();
+      f.Condition.Accept (this);
+      Out ($"    brfalse {lab2}");
+      Out ($"  {lab1}:");
+      f.IfPart.Accept (this);
+      Out ($"    br {lab3}");
+      Out ($"  {lab2}:");
+      f.ElsePart?.Accept (this);
+      Out ($"  {lab3}:");
+   }
```
Comment (tarydon) [inline PSICompile/ILCodeGen.cs:80]: This works, but uses too many labels. Let's take an if statement with no else part (for example like this):
`if (2 < 3) then WriteLn (2);`
You are generating this code:
```
    ldc.i4 2
    ldc.i4 3
    clt
    brfalse IL_0002
  IL_0001:
    ldc.i4 2
    call void [System.Console]System.Console::Write (int32)
    call void [System.Console]System.Console::WriteLine ()
    br IL_0003
  IL_0002:
  IL_0003:
```
I am expecting this code:
```
    ldc.i4 2
    ldc.i4 3
    clt
    brfalse IL_0000
    ldc.i4 2
    call void [System.Console]System.Console::Write (int32)
    call void [System.Console]System.Console::WriteLine ()
  IL_0001:
```

File: PSICompile/ILCodeGen.cs [hunk 3]
```diff
@@ -122,7 +132,10 @@ public class ILCodeGen : Visitor {
       u.Expr.Accept (this);
       string op = u.Op.Kind.ToString ().ToLower ();
       op = op switch { "sub" => "neg", _ => op };
-      Out ($"    {op}");
+      if (op is "not") {
+         Out ("    ldc.i4.0");
+         Out ("    ceq");
+      } else Out ($"    {op}");
```
Comment (tarydon) [inline PSICompile/ILCodeGen.cs:138]: This works correctly only for boolean values. So this works:
`WriteLn (not false)` should print **True.**
But this does not work:
`WriteLn (not 2)` should print **-3**
Because the not of an integer value is the integer you get by flipping all the bits in the binary representation. So the code you generate for not should depend on the data type of the expression.

File: PSICompile/ILCodeGen.cs [hunk 4]
```diff
@@ -131,11 +144,16 @@ public class ILCodeGen : Visitor {
          Out ("    call string [System.Runtime]System.String::Concat (string, string)");
       else {
          string op = b.Op.Kind.ToString ().ToLower ();
-         op = op switch { "mod" => "rem", "eq" => "ceq", "lt" => "clt", _ => op };
+         bool cmpeq = (op is "neq" or "geq" or "leq");
+         op = op switch { "mod" => "rem", "eq" or "neq" => "ceq", "lt" or "geq" => "clt", "gt" or "leq" => "cgt", _ => op };
```
Comment (tarydon) [inline PSICompile/ILCodeGen.cs:148]: You could have a multi-line string in the case expression output like:
`"geq" => "clt\n    ldc.i4.0\n    ceq"`
That might make this code cleaner. In principle, IL accepts even multiple op-codes on the same line, so you could even output just `clt  ldc.i4.0  ceq` on one line and it will work. However, that is non-conventional and I would discourage that.

**Review by tarydon — CHANGES_REQUESTED** [review]

File: PSICompile/ILCodeGen.cs (hunk 2 above)
Comment (SasiTMM) [reply PSICompile/ILCodeGen.cs:80]: Generated the code as above:
image

but there is a compiler error of missing IL_0000 in the forward reference.
image

correct me, if i am doing anything wrong!

File: PSICompile/ILCodeGen.cs (hunk 4 above)
Comment (SasiTMM) [reply PSICompile/ILCodeGen.cs:148]: updated the code with multi-line string as follows:
image

File: PSICompile/ILCodeGen.cs (hunk 2 above)
Comment (SasiTMM) [reply PSICompile/ILCodeGen.cs:80]: hope it's typo error at brfalse:
image

now updated the code as follows:
image

File: PSICompile/ILCodeGen.cs (hunk 3 above)
Comment (SasiTMM) [reply PSICompile/ILCodeGen.cs:138]: implemented "not" operator by checking the data type of expression:

image

File: Output/PSIOutput.il (hunk 1 above)
Comment (SasiTMM) [reply Output/PSIOutput.il:4]: Added the output folder in the .gitignore file
